modules/afsocket/afsocket-source.c

```diff
@@ -287,19 +288,19 @@ afsocket_sd_set_dynamic_window_size(LogDriver *s, gint dynamic_window_size)
 }
 
 void
-afsocket_sd_set_dynamic_window_stats_freq(LogDriver *s, gint stats_freq)
+afsocket_sd_set_dynamic_window_stats_freq(LogDriver *s, gfloat stats_freq)
 {
   AFSocketSourceDriver *self = (AFSocketSourceDriver *) s;
 
-  self->dynamic_window_stats_freq = stats_freq;
+  self->dynamic_window_stats_freq = stats_freq * 1000.0f;
```

Comment:
By `LL_FLOAT`, negative numbers can be set here, so I think we should give a stricter restriction in the grammar file, so it throws error at config load time.

```
@version: 3.20
options {
 min-iw-size-per-reader(1);
 stats_level(2);
};

source s_network_dynamic {
  network(ip(0.0.0.0)
    port(61001)
    log_fetch_limit(1000)
    log_iw_size(10000)
    max_connections(10000)
    dynamic-window-size(100000)
    dynamic-window-stats-freq(-0.1)
    dynamic-window-realloc-ticks(10)
    dynamic-window-decrease-threshold(50)
    dynamic-window-increase-threshold(20)
    flags(no-parse) 
  );
};

destination d_local {
	file("/tmp/ne_test.log");
};

log {
	source(s_network_dynamic);
	destination(d_local);
};
```

```
[2019-05-06T10:46:58.080088] Accepting connections; addr='AF_INET(0.0.0.0:61001)'
[2019-05-06T10:46:58.080254] Running application hooks; hook='1'
[2019-05-06T10:46:58.080274] Running application hooks; hook='6'
[2019-05-06T10:46:58.080312] syslog-ng starting up; version='3.20.1.350.gd19ec66'
[2019-05-06T10:46:58.080333] Running application hooks; hook='2'
[2019-05-06T10:46:58.080378] Dynamic window timer elapsed; tick='1'
[2019-05-06T10:46:58.080399] Dynamic window timer elapsed; tick='2'
[2019-05-06T10:46:58.080404] Dynamic window timer elapsed; tick='3'
[2019-05-06T10:46:58.080408] Dynamic window timer elapsed; tick='4'
[2019-05-06T10:46:58.080413] Dynamic window timer elapsed; tick='5'
[2019-05-06T10:46:58.080419] Dynamic window timer elapsed; tick='6'
[2019-05-06T10:46:58.080425] Dynamic window timer elapsed; tick='7'
[2019-05-06T10:46:58.080430] Dynamic window timer elapsed; tick='8'
[2019-05-06T10:46:58.080435] Dynamic window timer elapsed; tick='9'
[2019-05-06T10:46:58.080439] Dynamic window timer elapsed; tick='10'
[2019-05-06T10:46:58.080443] Dynamic window timer elapsed; tick='1'
[2019-05-06T10:46:58.080447] Dynamic window timer elapsed; tick='2'
[2019-05-06T10:46:58.080451] Dynamic window timer elapsed; tick='3'
[2019-05-06T10:46:58.080455] Dynamic window timer elapsed; tick='4'
[2019-05-06T10:46:58.080459] Dynamic window timer elapsed; tick='5'
[2019-05-06T10:46:58.080463] Dynamic window timer elapsed; tick='6'
[2019-05-06T10:46:58.080467] Dynamic window timer elapsed; tick='7'
[2019-05-06T10:46:58.080471] Dynamic window timer elapsed; tick='8'
[2019-05-06T10:46:58.080475] Dynamic window timer elapsed; tick='9'
[2019-05-06T10:46:58.080478] Dynamic window timer elapsed; tick='10'
```

Reply:
We could make a `positive_float` type, which accepts `positive_integer` or `LL_FLOAT` with validation. This way we do not need to write the rule in 2 lines. Same can be done with nonnegative, etc...